File: 8/cpp/main.cpp
#include <iostream>

class Solution {
public:
    int myAtoi(string s) {
        int sign = 1;
        int offset = 0;
        int n = s.length();
        
        // Ignore leading whitespace characters
        while (offset < n && s[offset] == ' ') {
            offset++;
        }
        
        // Handle sign if present
        if (s[offset] == '-') {
            sign = -1;
            offset++;
        }
        else if (s[offset] == '+') {
            offset++;
        }

        // Initialize result
        long long result = 0;

        // Process numeric characters
        while (offset < n && s[offset] >= '0' && s[offset] <= '9') {
            result = result * 10 + (s[offset] - '0');

            // Check for overflow/underflow
            if (result * sign > INT_MAX) {
                return INT_MAX;
            }
            if (result * sign < INT_MIN) {
                return INT_MIN;
            }

            offset++;
        }

        // Return the final result with the correct sign
        return static_cast<int>(result * sign);
    }
};

int main() {
    Solution sol;

    // Test case X: Y
    RETURN_TYPE resultX = sol.FUNCTION_NAME(Y);
    std::cout << "Test X: " << resultX << std::endl;

    return 0;
}

File: 8/python3/main.py
#!/usr/bin/env python3

class Solution:
    def myAtoi(self, s: str) -> int:
        # Initialize variables
        sign = 1
        result = 0
        i = 0
        n = len(s)
        INT_MAX = 2**31 - 1
        INT_MIN = -2**31

        # Skip leading whitespace
        while i < n and s[i] == ' ':
            i += 1

        # Handle sign if present
        if i < n and s[i] == '-':
            sign = -1
            i += 1
        elif i < n and s[i] == '+':
            i += 1

        # Process the number
        while i < n and s[i].isdigit():
            result = result * 10 + int(s[i])
            
            # Check for overflow/underflow
            if result * sign > INT_MAX:
                return INT_MAX
            if result * sign < INT_MIN:
                return INT_MIN
            
            i += 1

        # Return result with sign
        return result * sign
